nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -506,13 +506,17 @@ bool RegulatedPurePursuitController::isCollisionImminentExtendedSearch()
     global_plan_.poses.begin(),
     last_pose_it,
     std::back_inserter(extended_collision_check_path.poses),
-    [](geometry_msgs::msg::PoseStamped & poseStamped) {
+    [&](geometry_msgs::msg::PoseStamped & poseStamped) {
```

Comment:
Try passing only the objects you need 
`[global_plan_, costmap_ros_](geometry_msgs::msg::PoseStamped & poseStamped) `

or something like that